Source/WebKit/WebProcess/WebProcess.cpp

```diff
@@ -1677,12 +1681,29 @@ void WebProcess::nonVisibleProcessEarlyMemoryCleanupTimerFired()
     if (!m_pagesInWindows.isEmpty())
         return;
 
+    destroyDecodedImageData();
+
 #if PLATFORM(COCOA)
     destroyRenderingResources();
     releaseSystemMallocMemory();
 #endif
 }
 
+void WebProcess::destroyDecodedImageData()
+{
+    for (auto& page : m_pageMap.values())
+        page->releaseAllRemoteImageResources();
+
+    MemoryCache::singleton().destroyDecodedDataForAllImages();
+    BitmapImage::setRequireSynchronousRedecoding(true);
```

Comment:
What about the case where `<img decoding=async>`? I think we also want to make that decode synchronously in this case, otherwise we will see a flicker of unpainted image when we go from the page snapshot (which does show the image) to the next actual paint (when we start the async decode).

Reply:
Yes this is true and this what we do `RenderBoxModelObject::decodingModeForImageDraw()` if we passed the `PaintBehavior::TileFirstPaint` milestone and we are inside the viewport. It returns `DecodingMode::Synchronous`. So all what we need is to pretend the layer has been painted previously.

Reply:
So I think that means we'll decode async even when ignoring `TileFirstPaint`.